### PR TITLE
Fix Microsoft/vscode-chrome-debug#426: handle double dots in path

### PR DESCRIPTION
**Pull request by bpesquet**
I used a regexp to handle double dots and updated the test suite accordingly. 
Linting and tests are ok.

### CODE REVIEW

**Comment by roblourens** [comment]
Thanks for the PR! It looks like this isn't resolving the .., but just removing it, which wouldn't fix the original issue, right? And I'd much rather use `path.resolve` from Node, since it will handle single dots and all the edge cases.

File: test/chrome/chromeUtils.test.ts [hunk 1]
```diff
@@ -318,6 +318,27 @@ suite('ChromeUtils', () => {
                 chromeUtils.getMatchingTargets(targets, 'http://localhost'),
                 targets);
         });
+
+        test('handles double dots', () => {
+            const targets = makeTargets('http://localhost/app', 'http://localhost/site/../folder/../app');
```
Comment (roblourens) [inline test/chrome/chromeUtils.test.ts]: Thanks for writing tests, but the target won't have `..`, and these tests aren't actually testing that the `..` is resolved, right?

**Comment by bpesquet** [comment]
> It looks like this isn't resolving the .., but just removing it, which wouldn't fix the original issue, right?

Actually it removed the `..` *and* the parent folder, so that `localhost/site/../app` becomes `localhost/app`.

Anyway, I followed your advice and used `path.normalize()` instead. My local tests run fine, but some CI ones on AppVeyor fail. I may have missed something, my apologies if it is the case.

**Comment by roblourens** [comment]
I don't think we can apply path.normalize to everything, as it will break URLs:

```
> require('path').normalize('http://localhost')
'http:/localhost'
```

I'm not sure we should resolve .. in all URLs anyway. Just `file:///` URLs.

**Comment by roblourens** [comment]
I don't think we can apply path.normalize to everything, as it will break URLs:

```
> require('path').normalize('http://localhost')
'http:/localhost'
```

I'm not sure we should resolve .. in all URLs anyway. Just `file:///` URLs.

**Comment by roblourens** [comment]
Appveyor runs on windows, and if you send a URL through a `path` API without fixing it afterwards, you'll get backslashes back.

File: test/chrome/chromeUtils.test.ts (hunk 1 above)
Comment (bpesquet) [inline test/chrome/chromeUtils.test.ts]: Sorry, I don't understand what you mean by "resolve". A double dot in a path means going up one level in the folder structure: `my/path/to/this/../resource` resolves to `my/path/to/resource`. 

That's precisely my use case (`file:///Users/baptiste/Documents/Projets/GitHub/bpesquet/thejsway-code/chapter02/js/../html/celsius_fahrenheit.html` should resolve to `file:///Users/baptiste/Documents/Projets/GitHub/bpesquet/thejsway-code/chapter02/html/celsius_fahrenheit.html`), and I think that's what my code does.

Comment (roblourens) [inline test/chrome/chromeUtils.test.ts]: Actually I see, this is testing that it matches `targets[0]` and doesn't match the one with dots. Sorry, I misread it.
